src/components/ItemTable.tsx

```diff
@@ -1,51 +1,55 @@
-import "./TagStyle.css";
-import { DataGrid, GridColDef, GridRowsProp } from "@mui/x-data-grid";
 import { DataStore } from "@aws-amplify/datastore";
+import {
+  DataGrid,
+  GridColDef,
+  GridRenderCellParams,
+  GridValueFormatterParams,
+} from "@mui/x-data-grid";
 import { useEffect, useState } from "react";
-// import TagsList from "./TagsList";
-import { Item } from "../models";
+import { Category, CategoryItem, Item } from "../models";
+import "./TagStyle.css";
+import TagsList from "./TagsList";
 
-// Sample data, should remove it after the fetch call is fixed
-// REMARKS: the type of categorys here doesn't matched with the data
-const rows: GridRowsProp = [
+const columns: GridColDef[] = [
+  { field: "name", headerName: "Name", minWidth: 350 },
   {
-    id: 1,
-    name: "Air Jordan",
-    dataAdded: "2/14/2023",
-    value: 60,
-    image: "",
-    categorys: ["Clothing", "tag3", "tag4"],
+    field: "value",
+    headerName: "Price",
+    minWidth: 200,
+    flex: 0.5,
+    valueFormatter: (params: GridValueFormatterParams) => {
+      const value = params.value as number;
+      return `$${value.toFixed(2)}`;
+    },
   },
   {
-    id: 2,
-    name: "New balance sneaker",
-    dataAdded: "2/23/2023",
-    value: "$60",
-    image: "",
-    categorys: ["Alothing", "tag2", "tag3", "tag4", "tag5"],
+    field: "dateAdded",
+    headerName: "Date Added",
+    minWidth: 200,
+    flex: 0.5,
+    valueFormatter: (params: GridValueFormatterParams) => {
+      const date = new Date(params.value as string);
+      return date.toLocaleDateString();
+    },
   },
-];
-
-const columns: GridColDef[] = [
-  { field: "name", headerName: "Name", minWidth: 200 },
-  { field: "value", headerName: "Price", minWidth: 150 },
-  { field: "dataAdded", headerName: "Date Added", minWidth: 150 },
-  // TODO: put the Categorys back once the fetch call is fixed
-  /* {
-    field: "categorys",
-    headerName: "Categorys",
+  {
+    field: "categories",
+    headerName: "Categories",
     minWidth: 350,
     flex: 2,
-    renderCell: (params: GridRenderCellParams<Date>) => (
-      <strong>
-        <TagsList categories={params.value} />
-      </strong>
+    renderCell: (params: GridRenderCellParams) => (
+      <TagsList categories={params.value as string[]} />
     ),
-  }, */
+    sortComparator: (v1: string[], v2: string[]) => {
+      const v1Str = v1.join("");
+      const v2Str = v2.join("");
+      return v1Str.localeCompare(v2Str);
+    },
+  },
 ];
 
 export default function ItemTable() {
-  // const [rows, setRows] = useState<LazyItem[]>([]);
+  const [rows, setRows] = useState<any>([]);
```

Comment:
Will it be better if we declare the type of rows? I tried LazyItem or Item, both works.

Reply:
Yes! I put any to test out solutions and forgot to define the actual type. Thanks for catching this